shapes/produce_shapes_2016.py

```diff
@@ -620,6 +620,64 @@ def readclasses(c):
                     channel=tt,
                     era=era)
 
+    # Tau ID 
+    # in et and mt one nuisance per pT bin
+    # [30., 35., 40., 500., 1000. ,$\le$ 1000.]
+    for channel in ["et" , "mt"]:
+        pt = [30, 35, 40, 500, 1000, "inf"]
+        tau_id_variations = []
+        for i, ptbin in enumerate(pt[:-1]):
+            bindown = ptbin
+            binup = pt[i+1]
+            if binup == "inf":
+                tau_id_variations.append(
+                AddWeight("CMS_eff_tauid_{}-{}_{}_Run2016".format(bindown, binup, channel), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel),
+                        Weight("(((pt_2 >= {bindown})*tauIDScaleFactorWeightUp_tight_DeepTau2017v2p1VSjet_2*1.0/tauIDScaleFactorWeightUp_tight_DeepTau2017v2p1VSjet_2)+((pt_2 < {bindown})*1.0))".format(bindown=bindown), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel)), "Up"))
+                tau_id_variations.append(
+                    AddWeight("CMS_eff_tauid_{}-{}_{}_Run2016".format(bindown, binup, channel), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel),
+                            Weight("(((pt_2 >= {bindown})*tauIDScaleFactorWeightDown_tight_DeepTau2017v2p1VSjet_2*1.0/tauIDScaleFactorWeight_tight_DeepTau2017v2p1VSjet_2)+((pt_2 < {bindown})*1.0))".format(bindown=bindown), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel)), "Down"))
+            else:
+                tau_id_variations.append(
+                AddWeight("CMS_eff_tauid_{}-{}_{}_Run2016".format(bindown, binup, channel), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel),
+                        Weight("(((pt_2 >= {bindown} && pt_2 < {binup})*tauIDScaleFactorWeightDown_tight_DeepTau2017v2p1VSjet_2*(1.0/tauIDScaleFactorWeight_tight_DeepTau2017v2p1VSjet_2))+((pt_2 < {bindown} && pt_2 > {binup})*1.0))".format(bindown=bindown, binup=binup), "tauid_{}-{}_{}_eff_weight".format(bindown, binup, channel)), "Up"))
```

Comment:
Please fix

Reply:
oh wow
you are so right - who told you about this bug ?

Reply:
☝️

Reply:
updated